pkg/operator/api/schema/metrics.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2019 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"github.com/cortexlabs/cortex/pkg/lib/pointer"
+	"github.com/cortexlabs/cortex/pkg/lib/slices"
+)
+
+type RegressionStats struct {
+	Min         *float64 `json:"min"`
+	Max         *float64 `json:"max"`
+	Avg         *float64 `json:"avg"`
+	SampleCount int      `json:"sample_count"`
+}
+
+func (left RegressionStats) Merge(right RegressionStats) RegressionStats {
+	totalSampleCount := left.SampleCount + right.SampleCount
+
+	return RegressionStats{
+		Min:         slices.Float64PtrMin(left.Min, right.Min),
+		Max:         slices.Float64PtrMax(left.Max, right.Max),
+		Avg:         mergeAvg(left.Avg, left.SampleCount, right.Avg, right.SampleCount),
+		SampleCount: totalSampleCount,
+	}
+}
+
+type NetworkStats struct {
+	Latency *float64 `json:"latency"`
+	Code2XX int      `json:"code_2xx"`
+	Code4XX int      `json:"code_4xx"`
+	Code5XX int      `json:"code_5xx"`
+	Total   int      `json:"total"`
+}
+
+func (left NetworkStats) Merge(right NetworkStats) NetworkStats {
+	return NetworkStats{
+		Latency: mergeAvg(left.Latency, left.Total, right.Latency, right.Total),
+		Code2XX: left.Code2XX + right.Code2XX,
+		Code4XX: left.Code4XX + right.Code4XX,
+		Code5XX: left.Code5XX + right.Code5XX,
+		Total:   left.Total + right.Total,
+	}
+}
+
+type APIMetrics struct {
+	NetworkStats      *NetworkStats    `json:"network_stats"`
+	ClassDistribution map[string]int   `json:"class_distribution"`
+	RegressionStats   *RegressionStats `json:"regression_stats"`
+}
+
+func (left APIMetrics) Merge(right APIMetrics) APIMetrics {
+	mergedClassDistribution := left.ClassDistribution
+
+	if right.ClassDistribution != nil {
+		if left.ClassDistribution == nil {
+			mergedClassDistribution = right.ClassDistribution
+		} else {
+			for className, count := range right.ClassDistribution {
+				if _, ok := mergedClassDistribution[className]; ok {
```

Comment:
You might not like this idea, so up to you: I believe that since 0 is the default value in `APIMetrics.ClassDistribution`, you can remove the if check here and blindly do `mergedClassDistribution[className] += count`